modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -205,19 +206,22 @@ func (h Handler) copyResponse(dst http.ResponseWriter, src io.Reader, flushInter
 
 	buf := streamingBufPool.Get().(*[]byte)
 	defer streamingBufPool.Put(buf)
-	_, err := h.copyBuffer(w, src, *buf)
+	_, err := h.copyBuffer(w, src, *buf, logger)
 	return err
 }
 
 // copyBuffer returns any write errors or non-EOF read errors, and the amount
 // of bytes written.
-func (h Handler) copyBuffer(dst io.Writer, src io.Reader, buf []byte) (int64, error) {
+func (h Handler) copyBuffer(dst io.Writer, src io.Reader, buf []byte, logger *zap.Logger) (int64, error) {
```

Comment:
The logger is already stored in:
https://github.com/caddyserver/caddy/blob/1b73e3862d312ac2057265bf2a5fd95760dbe9da/modules/caddyhttp/reverseproxy/reverseproxy.go#L213

any reason why not use it as `h.logger.[...]`?

Reply:
This logger has added information like the request, which we found was really important for adding context so we could form a cohesive story. Otherwise we found ourselves asking, "Is this from OUR request or another one?"